clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/AbstractKafkaConsumerConfigProperties.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright (c) 2021 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.client.kafka.consumer;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.common.serialization.Deserializer;
+import org.eclipse.hono.client.kafka.AbstractKafkaConfigProperties;
+
+/**
+ * Configuration properties for Kafka consumers.
+ * <p>
+ * This class is intended to be as agnostic to the provided properties as possible in order to be forward-compatible
+ * with changes in new versions of the Kafka client.
+ */
+public abstract class AbstractKafkaConsumerConfigProperties extends AbstractKafkaConfigProperties {
+
+    /**
+     * The default amount of time (milliseconds) to wait for records when polling records.
+     */
+    public static final long DEFAULT_POLL_TIMEOUT = 100L; // ms
+
+    @SuppressWarnings("rawtypes")
+    private final Class<? extends Deserializer> keyDeserializerClass;
+
+    @SuppressWarnings("rawtypes")
+    private final Class<? extends Deserializer> valueDeserializerClass;
+
+    private Map<String, String> consumerConfig;
+    private long pollTimeout = DEFAULT_POLL_TIMEOUT;
+
+    /**
+     * Creates an instance.
+     *
+     * @param keyDeserializerClass The class to be used for serializing the record keys, if {@code null} the
+     *            deserializer needs to be provided in the configuration at runtime.
+     * @param valueDeserializerClass The class to be used for serializing the record values, if {@code null} the
+     *            deserializer needs to be provided in the configuration at runtime.
+     */
+    @SuppressWarnings("rawtypes")
+    protected AbstractKafkaConsumerConfigProperties(final Class<? extends Deserializer> keyDeserializerClass,
```

Comment:
I guess you want to pass in the type itself instead of just the class name to make sure that it actually is a Deserializer, right?
If so, I believe we can use `final Class<? extends Deserializer<?>> keyDeserializerClass)` and remove the `@SuppressWarning` annotations ...